apps/src/sites/studio/pages/levelbuilder_gamelab.js

```diff
@@ -33,25 +34,12 @@ $(document).ready(function () {
 
   // Leniently validate and fix up custom block JSON using jsonic
   if (document.getElementById('level_custom_blocks')) {
-    const customBlocksValidationDiv = $('#custom-blocks-validation');
-    const customBlocksEditor =
-      initializeCodeMirror('level_custom_blocks', 'application/json');
-    customBlocksEditor.on('blur', () => {
-      try {
-        if (customBlocksEditor.getValue().trim()) {
-          let blocks = jsonic(customBlocksEditor.getValue().trim());
-          if (!Array.isArray(blocks)) {
-            blocks = [blocks];
-          }
-          customBlocksEditor.setValue(JSON.stringify(blocks, null, 2));
-        } else {
-          customBlocksEditor.setValue('');
+    initializeCodeMirrorForJson('level_custom_blocks', {
+      validationDiv: 'custom-blocks-validation',
+      onBlur(blocks) {
+        if (!Array.isArray(blocks)) {
+          return [blocks];
         }
```

Comment:
is it intentional that this doesn't do anything if `blocks` _is_ an array?

Reply:
It is not! Thanks for catching that.